app/models/dynamic_annotation/field.rb

```diff
@@ -62,4 +64,10 @@ def set_field_type
   def set_json_value
     self.value_json = self.value if self.field_type == 'json' && self.respond_to?(:value_json)
   end
+
+  def set_author_slack_channel_url
+    a = self.annotation.load
+    value = a.get_field('smooch_user_data')&.value_json
+    Rails.cache.write("SmoochUserSlackChannelUrl:#{a.annotated_type}:#{a.annotated_id}:#{value['id']}", self.value) unless value.nil?
+  end
```

Comment:
Sawy, for this stuff that is specific for the Smooch Bot, we prefer to extend the class in the Smooch Bot class instead of adding custom logic to the core class... please add that logic to this place: https://github.com/meedan/check-api/blob/develop/app/models/bot/smooch.rb#L100